### PR TITLE
fix(atomic/commerce): facet / facetState mismatch when rendering generated facets

### PR DESCRIPTION
**Pull request by fbeaudoincoveo**
https://coveord.atlassian.net/browse/KIT-3366

### CODE REVIEW

File: packages/atomic/src/components/commerce/facets/atomic-commerce-facets/atomic-commerce-facets.tsx [hunk 1]
```diff
@@ -104,6 +104,7 @@ export class AtomicCommerceFacets implements InitializableComponent<Bindings> {
             summary: this.summary,
             facet: facet as T,
             field: facet.state.field,
+            key: facet.state.facetId,
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/commerce/facets/atomic-commerce-facets/atomic-commerce-facets.tsx:107]: As mentioned in the Jira description (https://coveord.atlassian.net/browse/KIT-3366), this is the actual fix. The remaining changes in this PR are mostly for the sake of uniformity.

**Review by Spuffynism — APPROVED** [review]

**Review by alexprudhomme — APPROVED** [review]

**Review by olamothe — APPROVED** [review]
Since we now have playwright facet tests in place, anything that could be added to the existing test suite to catch this in the future ?

**Comment by fbeaudoincoveo** [comment]
> Since we now have playwright facet tests in place, anything that could be added to the existing test suite to catch this in the future ?

Indeed I believe it would be possible to add an e2e test that covers that 👍

**Comment by fbeaudoincoveo** [comment]
> Since we now have playwright facet tests in place, anything that could be added to the existing test suite to catch this in the future ?

@olamothe I created a technical debt Jira, because I believe in order to write these tests we'll need our own testing environment: https://coveord.atlassian.net/browse/KIT-3372.

I'll merge this PR and tackle the tech debt one as soon as possible.

**Pull request merged**